libs/ui/src/FileList/FileListItem.tsx

```diff
@@ -1,61 +1,62 @@
 import { Button } from '@amsterdam/design-system-react/dist/Button'
-import useIsAfterBreakpoint from '@amsterdam/design-system-react/dist/common/useIsAfterBreakpoint'
-import { ErrorMessage } from '@amsterdam/design-system-react/dist/ErrorMessage'
+import { Icon } from '@amsterdam/design-system-react/dist/Icon'
 import { Image } from '@amsterdam/design-system-react/dist/Image'
-import { clsx } from 'clsx'
+import { WarningIcon } from '@amsterdam/design-system-react-icons'
+import clsx from 'clsx'
 import { HTMLAttributes, useMemo } from 'react'
 
-import { formatFileSize } from './formatFileSize'
-
 import styles from './FileList.module.css'
 
 // Although a description list (<dl>, <dt> and <dd>) would be more semantically correct than an unordered list (<ul> and <li>),
 // an unordered list with list items is used here because NVDA currently (16-9-2025) reads the number of items in a description list incorrectly.
 
 export type FileListItemProps = HTMLAttributes<HTMLLIElement> & {
   deleteButtonId: string
-  deleteButtonLabel?: string
   errorMessage?: string
   file: File
   onDelete?: () => void
+  progress: number
+  status: 'pending' | 'uploading' | 'success' | 'error'
 }
 
-export const FileListItem = ({
-  deleteButtonId,
-  deleteButtonLabel = 'Verwijder',
-  errorMessage,
-  file,
-  onDelete,
-}: FileListItemProps) => {
+export const FileListItem = ({ deleteButtonId, errorMessage, file, onDelete, progress, status }: FileListItemProps) => {
   // Memoize the creation of an object url from the file,
   // to prevent it from creating a new one on every render.
   const imageUrl = useMemo(() => URL.createObjectURL(file), [file])
 
+  const actionButtonLabel = status === 'success' || status === 'error' ? 'Verwijderen' : 'Annuleren'
+  const progressPercentage = status === 'success' ? 'geslaagd' : Math.round(progress) + '%'
+
+  const statusMessage =
+    status === 'error' ? (
+      <span className={styles.errorMessage}>{errorMessage}</span>
+    ) : (
+      <span className={styles.statusMessage}>{`Upload ${progressPercentage}`}</span>
+    )
+
+  const image =
+    status === 'error' ? (
+      <Icon svg={WarningIcon} size="heading-1" className={styles.icon} />
+    ) : (
+      <Image src={imageUrl} alt="" />
```

Comment:
It's useful to set a default width on this image which you override in CSS, in order to prevent images from blowing up when CSS doesn't load. See the old code for an example.